### ISSUE TITLE
`istioctl manifest migrate` produces an large, unwieldy yaml file

### ISSUE THREAD

**Issue by XanderStrike** [istio/istio#19501]
**Bug description**

Against a cluster that has been set up using helm and a small ~40 line values file, `istioctl manifest generate` produces a 1.6k line yaml file.

<details>
  <summary>values.yaml</summary>

```yaml
gateways:
  istio-ingressgateway:
    autoscaleMin: 20
    autoscaleMax: 20
    tolerations:
    - key: "scalers.istio"
      operator: "Equal"
      value: "dedicated"
      effect: "NoSchedule"
    nodeSelector:
      scalers.istio: dedicated
    podAntiAffinityTermLabelSelector:
    - key: istio
      operator: In
      values: ingressgateway
      topologyKey: "kubernetes.io/hostname"

pilot:
  autoscaleMin: 20
  autoscaleMax: 20
  tolerations:
  - key: "scalers.istio"
    operator: "Equal"
    value: "dedicated"
    effect: "NoSchedule"
  nodeSelector:
    scalers.istio: dedicated
  podAntiAffinityTermLabelSelector:
  - key: istio
    operator: In
    values: pilot
    topologyKey: "kubernetes.io/hostname"

mixer:
  policy:
    enabled: false
  telemetry:
    enabled: false
```
</details>

[Here](https://github.com/istio/istio/files/3946824/manifest-migrate.yaml.txt) is the result with `--logtostderr` (which makes it log to stdout...???) enabled.

Note also the following error which is present in the log:

```
error translating values.yaml: error when unmarshalling into control plane spec unknown field "key" in v1.WeightedPodAffinityTerm
```
This does not appear in our `values.yaml` and seems to indicate that the migration failed.

And finally [here](https://github.com/istio/istio/files/3946834/helm-generated.yaml.txt) is our generated helm chart, but it's just created by running our `values.yaml` against 1.4.0.

**Desired behavior**

It'd be nice if all that was returned was a diff between my deployment and the default (or another chosen profile) so that it would be easier to see the changes that we have made, and to continue working with this file moving forward.

In other words, we want what a human would have written to _modify_ their installation, rather than a complete manifest.

**Comment by richardwxn** [comment]
Did you just run istioctl manifest migrate command without any extra arguments in the cluster? Also would you mind adding the link to the command output?

**Comment by XanderStrike** [comment]
Yes, run without any arguments. The command output is in the issue.

**Comment by ostromart** [comment]
error translating values.yaml: error when unmarshalling into control plane spec unknown field "key" in v1.WeightedPodAffinityTerm, 
seems to be the problem.

**Comment by ostromart** [comment]
@XanderStrike I agree a small input file should produce a small output file. We definitely need to investigate and fix.

**Comment by richardwxn** [comment]
Did some investigation and it seems like there is issue about the code to translate the node and pod affinity settings. Will create a PR soon.

**Comment by richardwxn** [comment]
fixed in master and 1.5 already
